retrofit-adapters/rxjava2/src/main/java/retrofit2/adapter/rxjava2/NoContentException.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2016 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package retrofit2.adapter.rxjava2;
+
+import io.reactivex.Completable;
+import retrofit2.Response;
+
+/** Exception for an unexpected, 204/205 HTTP response when not used with {@link Completable}. */
+public final class NoContentException extends RuntimeException {
+  private static String getMessage(Response<?> response) {
+    if (response == null) {
+      throw new NullPointerException("response == null");
+    }
+    return "HTTP " + response.code() + " " + response.message();
+  }
+
+  private final int code;
+  private final String message;
+  private final transient Response<?> response;
+
+  public NoContentException(Response<?> response) {
+    super(getMessage(response));
+    this.code = response.code();
+    this.message = response.message();
+    this.response = response;
+  }
+
+  /** HTTP status code. */
+  public int code() {
+    return code;
+  }
+
+  /** HTTP status message. */
+  public String message() {
+    return message;
+  }
+
+  /**
+   * The full HTTP response. This may be null if the exception was serialized.
+   */
+  public Response<?> response() {
```

Comment:
Needs `@Nullable`

Reply:
Should `HttpException` (where this is copied from) be updated too?

Reply:
`HttpException` was updated and that test ensures you can't supply `null`, not that it can't return `null`. Note the Javadoc which details when you will receive a `null` value.

Reply:
Ahh I missed the `transient` modifier. Got it d90cbe4d8f0ec5e88ce56d1de28ca7b4483da0d2